lib/storage/index.js

```diff
@@ -26,146 +26,144 @@ const validDataUrl = require('valid-data-url');
+                throw new SourceUnsupportedError(`Invalid or missing local file ${source.path}`);
+            }
+            if (disableDownload) {
+                console.log(`Skipping source file download for ${AssetComputeLogUtils.redactUrl(source.url)}`);
+                const preSignedUrl = await datauri.getPreSignedUrl(source.path);
+                console.log(`Uploaded data URI content to storage and generated presigned url`);
+                assetReference.url = preSignedUrl;
+                return new Asset(assetReference, directory, name);
+            }
+        } else if (!disableDownload && protocol === 'https:') {
+            console.log("downloading asset:", source.path);
+            await http.download(assetReference, source.path);
+        }
+    
+        return source;
     }
-    return ext;
-}
-
-/**
- * Returns if the file exists and is not empty
- *
- * @param {Number} path location of the file
- * @returns {Boolean} Returns false if file does not exist or is empty
- */
-function fileExistsAndIsNotEmpty(path) {
-    if(path && fs.existsSync(path)){
-        const fileStats =  fs.statSync(path);
-        if(fileStats.size !== 0){
-            return true;
+    
+    // function to return an extension for a file
+    // if not empty returns a leading period
+    // prefers extension from the file over name determined by mimeType
+    static extension(filename, mimeType) {
+        let ext = '';
+        if (filename) {
+            ext = path.extname(filename);
         }
+        if (!ext && mimeType) {
+            const mimeExt = mime.extension(mimeType);
+            ext = mimeExt ? `.${mimeExt}` : '';
+        }
+        return ext;
     }
-    return false;
-}
-
-// There is at least one worker (graphics magick) that in some cases depends
-// upon the file extension so it is best to try to use the appropriate one
-// based on the filename, url, or mimetype
-function sourceFilename(source) {
-    if (source.name) {
-        return `${SOURCE_BASENAME}${extension(source.name, source.mimeType)}`;
+    
+    /**
+     * Returns if the file exists and is not empty
+     *
+     * @param {Number} path location of the file
+     * @returns {Boolean} Returns false if file does not exist or is empty
+     */
+    static fileExistsAndIsNotEmpty(path) {
+        if(path && fs.existsSync(path)){
+            const fileStats =  fs.statSync(path);
+            if(fileStats.size !== 0){
+                return true;
+            }
+        }
+        return false;
     }
-
-    if (source.url && validUrl.isUri(source.url)) {
-        const basename = path.basename(new URL(source.url).pathname);
-        return  `${SOURCE_BASENAME}${extension(basename, source.mimeType)}`;
+    
+    // There is at least one worker (graphics magick) that in some cases depends
+    // upon the file extension so it is best to try to use the appropriate one
+    // based on the filename, url, or mimetype
+    static sourceFilename(source) {
```

Comment:
Either extension is a property or a function.
If a property, it should be static get sourceFilename(_, _).
If a function, it should be called getSourceFilename.

I would go for the property.